tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs

```diff
@@ -24,19 +26,162 @@ public static class TranscriptUtilities
         /// <returns>A list of activities to test</returns>
         public static IEnumerable<IActivity> GetFromTestContext(TestContext context)
         {
-            var transcriptsRootFolder = TestUtilities.GetKey("TranscriptsRootFolder") ?? @"..\..\..\..\..\transcripts";
-            var directory = Path.Combine(transcriptsRootFolder, context.FullyQualifiedTestClassName.Split('.').Last());
-            var fileName = $"{context.TestName}.transcript";
-            var path = Path.Combine(directory, fileName);
+            var relativePath = Path.Combine(context.FullyQualifiedTestClassName.Split('.').Last(), $"{context.TestName}.chat");
+            return GetActivities(relativePath);
+        }
+
+        public static IEnumerable<IActivity> GetActivities(string relativePath)
+        {
+            var transcriptsRootFolder = TranscriptUtilities.EnsureTranscriptsDownload();
+            var path = Path.Combine(transcriptsRootFolder, relativePath);
+            if (!File.Exists(path))
+            {
+                path = Path.Combine(transcriptsRootFolder, relativePath.Replace(".chat", ".transcript", StringComparison.InvariantCultureIgnoreCase));
+            }
             if (!File.Exists(path))
             {
                 Assert.Fail($"Required transcript file '{path}' does not exists in '{transcriptsRootFolder}' folder. Review the 'TranscriptsRootFolder' environment variable value.");
             }
 
-            var content = File.ReadAllText(path);
-            return JsonConvert.DeserializeObject<Activity[]>(content);
+            string content;
+            if (string.Equals(path.Split('.').Last(), "chat", StringComparison.InvariantCultureIgnoreCase))
+            {
+                content = Chatdown(path);
+            }
+            else
+            {
+                content = File.ReadAllText(path);
+            }
+
+            var activities = JsonConvert.DeserializeObject<List<Activity>>(content);
+
+            var lastActivity = activities.Last();
+            if (lastActivity.Text.Last() == '\n')
+            {
+                lastActivity.Text = lastActivity.Text.Remove(lastActivity.Text.Length - 1);
+            }
+
+            return activities.Take(activities.Count - 1).Append(lastActivity);
         }
 
+        private static readonly object syncRoot = new object();
+        private static string TranscriptsTemporalPath { get; set; }
+
```

Comment:
The name "temporal" implies the file travels in time. IF so, that's awesome!!! If not... we should consider changing the name.